app/models/post.rb

```diff
@@ -1,2 +1,20 @@
 class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments
+  has_many :likes
+
+  after_create :increment_user_posts_count
+  before_destroy :update_post_counter_before_destroy
+
+  def recent_comments(limit = 5)
+    comments.order(created_at: :desc).limit(limit)
+  end
+
+  def increment_user_posts_count
+    author.increment!(:postcounter)
+  end
+
+  def update_post_counter_before_destroy
+    author.decrement!(:postcounter)
+  end
```

Comment:
- [ ] In your `Post` model, consider making the `increment_user_posts_count` and `update_post_counter_before_destroy` methods private. By marking them as private, you signify that these methods are meant for internal use within the `Post` model and should not be accessed or called from outside the class. This encapsulation helps maintain clean and organized code, preventing unintended usage and potential issues in the future.

```suggestion
private 

  def increment_user_posts_count
    author.increment!(:postcounter)
  end

  def update_post_counter_before_destroy
    author.decrement!(:postcounter)
  end
```